docs: Update CONTRIBUTING_DOCS.md (#402)
nginx/documentation: CONTRIBUTING_DOCS.md

### How to format internal links

Internal links should use Hugo shortcodes [ref](https://gohugo.io/methods/shortcode/ref/#article) (for absolute paths) and [relref](https://gohugo.io/methods/shortcode/relref/#article) (for relative paths).
Please note that we favor absolute paths, as these are easier to maintain.

> - Although file extensions are optional for Hugo, we include them as best practice for page anchors.
> - Relative paths are preferred, but just the filename is permissible.

Reviewer: ```suggestion
- Although file extensions (such as `.md`) are optional for Hugo, we include them as best practice for page anchors.
- We prefer relative paths.
```
